### PR TITLE
Added build log and jacoco report generation/uploading

### PR DESCRIPTION
**Pull request by jaydee029**
For issue galasa-dev/projectmanagement#1965

(Re-Raised)

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -86,14 +87,29 @@ jobs:
...
+        
+      - name: Upload Gradle Build Log
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: gradle-build-log
+          path: build.log
+          
+      - name: Upload Jacoco Report
+        if: always()
```
Comment (jadecarino) [inline .github/workflows/build.yaml:107]: After discussing with a colleague, we determined that we should only do these on failure of the workflow, as you wouldn't bother to check the log if the run is successful. Please can we update these to `if: failure()` and set a retention time of 7 days?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -86,14 +87,29 @@ jobs:
...
+        uses: actions/upload-artifact@v4
+        with:
+          name: jacoco-report
+          path: ${{github.workspace}}/galasa-managers-parent/build/reports/jacoco/jacocoMerge
```
Comment (jadecarino) [inline .github/workflows/build.yaml:111]: My colleague suggested a way to do this to sweep up all `.html` reports for each project which I think is more effective than using the jacocoMerge. Can you please update this to:
```
      - name: Upload Jacoco Reports as artifact
        if: failure()
        uses: actions/upload-artifact@v3
        with:
          name: unit-test-reports
          path: /home/runner/work/managers/managers/galasa-managers-parent/**/**/build/reports/**/*.html
          retention-days: 7
          if-no-files-found: ignore
```

**Review by jadecarino — CHANGES_REQUESTED** [review]
Hi Dhruv - my colleague and I discussed some things this morning so based on that I'd like you to make the requested changes and also add the upload of the Gradle build log and unit test reports to the PR build as well as the main build please. If someone in the team is trying to open a PR and their PR build fails, they will want access to these artefacts too, more so than someone monitoring the Main builds.

Please make these changes for the Managers build and I'll add this same comment on to other workflows with open PRs. The completed ones without open PRs we can handle later.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (jaydee029) [reply .github/workflows/build.yaml]: done

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by jaydee029** [comment]
> Hi Dhruv - my colleague and I discussed some things this morning so based on that I'd like you to make the requested changes and also add the upload of the Gradle build log and unit test reports to the PR build as well as the main build please. If someone in the team is trying to open a PR and their PR build fails, they will want access to these artefacts too, more so than someone monitoring the Main builds.
> 
> Please make these changes for the Managers build and I'll add this same comment on to other workflows with open PRs. The completed ones without open PRs we can handle later.

done

**Review by jadecarino — APPROVED** [review]
Approved for building

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**